### PR TITLE
Weld5

### LINKED ISSUE #19748: Reconsume Weld 5.0.0-Alpha2
https://repo1.maven.org/maven2/org/jboss/weld/weld-osgi-bundle/5.0.0.Alpha2/weld-osgi-bundle-5.0.0.Alpha2.jar

### PR DESCRIPTION
**Pull request by tevans78**
- Pull in Weld 5.0.0.Alpha2 and CDI API 4.0.0.Beta3
- Repeat AnnotationsTests with CDI 4.0
- Add BeanParser service to allow existing tests to pass on CDI 4.0

### CODE REVIEW

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.cdi.annotations_fat/bnd.bnd [hunk 1]
```diff
@@ -21,7 +21,9 @@ tested.features:\
 	connectors-2.0,\
 	servlet-4.0,\
 	servlet-5.0,\
-	cdi-3.0
+	cdi-3.0,\
+        servlet-6.0,\
+        cdi-4.0
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.annotations_fat/bnd.bnd:26]: Whitespace here is off.

File: dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * These tests look for changes in the FacesUrlTransformer & ConversationAwareViewHandler classes.
+ * If the checksum changes, then the latest classes should investigated and ported over to com.ibm.ws.jsfContainer
+ * See issue https://github.com/OpenLiberty/open-liberty/issues/14524
+ */
+public class WeldJSFCheckSumTest {
+
+    File bundle;
+
+    //TODO: this will all need to be updated once the EE10 version of JSF has been done
+    String bundleFileName = "io.openliberty.org.jboss.weld5.jar";
```
Comment (Azquelt) [inline dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java:34]: What's the current situation with this test? Which version of the classes is it actually checking?

**Review by Azquelt — APPROVED** [review]
I think this looks ok. We might want to re-think some of the duplicated code.

Previously we had several different bundles for cdi-1.2 and 2.0. 3.0 was just a transformed 2.0 but with 4.0, we'll start to have more classes being different again which won't necessarily be the same classes that were different between 1.2 and 2.0.

File: dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java (hunk 2 above)
Comment (Azquelt) [reply dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java:34]: and particularly, what does the TODO mean?

Comment (tevans78) [reply dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java:34]: This test needs to be updated once the new version of JSF has been implemented (rebased on the new version of Weld). For the moment all of the tests are disabled (`@Ignore`)

Comment (Azquelt) [reply dev/io.openliberty.org.jboss.weld5/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java:34]: ah, so they are :+1:

**Comment by tevans78** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#libby

**Pull request merged**